account/middleware.py

```diff
@@ -51,3 +62,32 @@ def process_request(self, request):
             if account:
                 tz = settings.TIME_ZONE if not account.timezone else account.timezone
                 timezone.activate(tz)
+
+
+class ExpiredPasswordMiddleware(object):
+
+    def process_request(self, request):
+        if request.user.is_authenticated() and not request.user.is_staff:
+            next_url = resolve(request.path).url_name
+            # Authenticated users must be allowed to access
+            # "change password" page and "log out" page.
+            # even if password is expired.
+            if next_url not in [settings.ACCOUNT_PASSWORD_CHANGE_REDIRECT_URL,
+                                settings.ACCOUNT_LOGOUT_URL,
+                                ]:
+                if check_password_expired(request.user):
+                    signals.password_expired.send(sender=self, user=request.user)
+                    messages.add_message(
+                        request,
+                        messages.WARNING,
+                        _("Your password has expired. Please save a new password.")
+                    )
+                    redirect_field_name = "next"  # fragile!
```

Comment:
This redirect field name reference ("next") is quite fragile because if the project uses a different string this will break. Note that the redirect field name in classes protected by `LoginRequiredMixin` and methods protected by `login_required` default to "next" but can be overridden by class attribute or by invocation parameter. However this code is middleware with no way to set/reset the redirect field name.

To resolve this fragility I suggest using an `account` global, something like `ACCOUNT_REDIRECT_FIELD_NAME`.